src/main/kotlin/com/dobby/backend/application/service/ExperimentPostService.kt

```diff
@@ -39,4 +50,26 @@ class ExperimentPostService(
             else -> throw IllegalArgumentException("Invalid input type: ${input::class.java.simpleName}")
         }
     }
+
+    fun validateFilter(input: GetExperimentPostsUseCase.Input) {
+        val locationInfo = input.customFilter.locationTarget
+
+        if (locationInfo?.areas != null) {
+            val isAll = locationInfo.areas.any { it.isAll() }
+
+            if (isAll && locationInfo.areas.size != 1)
+                throw ExperimentAreaSelectionException()
+
+            if (locationInfo.areas.size > 5)
+                throw ExperimentAreaOverflowException()
+
+            val selectedRegion = locationInfo.region
+            if (selectedRegion != null) {
+                val validAreas = Area.findByRegion(selectedRegion).map { it.name }
+                if (locationInfo.areas.map { it.name }.any { it !in validAreas }) {
+                    throw ExperimentAreaInCorrectException()
+                }
+            }
+        }
+    }
```

Comment:
유효성 검증 로직을 서비스 계층에서 처리하여 SRP 원칙을 잘 지킨 점이 정말 좋습니다! 👍 

여러 if 문이 중첩되어 있는데 이를 별도의 메서드로 추출하면 가독성도 높아지고, 각 검증을 독립적으로 관리할 수 있어 더 명확한 코드가 될 것 같습니다. 아래와 같이 추출하는 방법을 고려해볼 수 있을 것 같아요!

```suggestion
    fun validateFilter(input: GetExperimentPostsUseCase.Input) {
        val locationInfo = input.customFilter.locationTarget ?: return

        locationInfo.areas?.let {
            validateLocationAreas(it)
        }

        locationInfo.region?.let {
            validateRegion(locationInfo, it)
        }
    }

    private fun validateLocationAreas(areas: List<Area>) {
        if (areas.size > 5) throw ExperimentAreaOverflowException()

        if (areas.any { it.isAll() } && areas.size != 1) {
            throw ExperimentAreaSelectionException()
        }
    }

    private fun validateRegion(locationInfo: GetExperimentPostsUseCase.LocationTargetInput, region: Region) {
        val validAreas = Area.findByRegion(region).map { it.name }

        if (locationInfo.areas?.map { it.name }?.any { it !in validAreas } == true) {
            throw ExperimentAreaInCorrectException()
        }
    }
```